extension-androidauto/build.gradle.kts

```diff
@@ -23,7 +23,8 @@ android {
 }
 
 dependencies {
-  api(Dependencies.androidAutoMapboxMapSdk)
```

Comment:
we should not do this. we can adopt the new api downstream but we should not need to change dependency here

but yeah it's necessary for testing, just flagging that we do not want to merge it

Reply:
If you revert this, we should be able to see what the "upgrade experience" will look like when building the `android-auto-app`. one difference is that development builds do not have a version attached so gradle does not know how to resolve the version conflicts

Reply:
to make this easier, it may make sense to remove CompassWidget and LogoWidget from the extension. That way the upgrades will continue to work

Reply:
I've reverted the changes to the Android auto extension, and made the changes to widget APIs backwards compatible, this way it shouldn't break the compatibility between Maps SDK and Android Auto extension.

But I do wonder what shall we do in these cases, where a breaking change can still happen to the experimental APIs in the Maps SDK and it will potentially break the compatibility between Maps SDK and Android Auto extension.

Reply:
Related convo - https://github.com/mapbox/mapbox-maps-android/pull/1767#issuecomment-1298256927. I'd vote for aligning versions of the Maps SDK and the Auto Extension, and require to use same version across all the libraries.

Reply:
After validating there's no breaking behaviour to the Android Auto Extension/App, the dependency is reverted back to the original state in https://github.com/mapbox/mapbox-maps-android/pull/1782/commits/76e2ac20d18daa30d2c9598b32709047766af50b.

Regarding the versioning of the Android Auto extension, the discussion is happening offline.